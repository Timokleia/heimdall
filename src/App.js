import React, { Component, Fragment } from "react";
import * as Sentry from "@sentry/browser";
import { Redirect, Route, BrowserRouter as Router } from "react-router-dom";
import Login from "./components/views/public/Login";
import Logout from "./components/views/public/Logout";
import Forbiden from "./components/views/public/Forbiden";
import FundationList from "./components/views/Fundations/FundationList";
import FundationDetails from "./components/views/Fundations/FundationDetails";
import Transferts from "./components/views/tranferts/transferts";
import Dashboard from "./components/views/dashboard/dashboard";
import { Col, Nav, Row, Tab } from "react-bootstrap";
import { ToastContainer } from "react-toastify";
import MyNavbar from "./components/views/navbar";
import Homepage from "./components/views/homepage";
import { init as initApm } from "@elastic/apm-rum";
import { ApmRoute } from "@elastic/apm-rum-react";
import UserDashboard from "./components/views/users/userDashboard";
import Support from "./components/views/support/support";
import SystemHomepage from "./components/views/SystemHomepage";
import "./App.css";
import packagejson from "../package.json";
import heimdalConfig from "./config";
import ComponentWrapper from "./components/ComponentWrapper";

import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faAmbulance,
  faBuilding,
  faHandHoldingUsd,
  faKey,
  faSuitcase,
  faTrafficLight,
  faUser
} from "@fortawesome/free-solid-svg-icons";

Sentry.init({
  dsn: heimdalConfig.SENTRY_DSN,
  environment: process.env.NODE_ENV,
  release: `${packagejson.name}@${packagejson.version}`
});

Sentry.configureScope(function(scope) {
  scope.setTag("NEMOPAY_VERSION", heimdalConfig.NEMOPAY_VERSION);
});

const apm = initApm(heimdalConfig.APM);

library.add(
  faBuilding,
  faSuitcase,
  faKey,
  faHandHoldingUsd,
  faUser,
  faTrafficLight,
  faAmbulance
);

class App extends Component {
  render() {
    const isLoggedIn = localStorage.hasOwnProperty("accessToken");
    const username = isLoggedIn
      ? JSON.parse(localStorage.getItem("accessToken")).username
      : null;
    apm.setUserContext({ username });
    return (
      <Fragment key='App'>
        <div className='App'>
          <Router>
            <ComponentWrapper
              pathSuffix='forbiden'
              exact
              navbar
              component={Forbiden}
            />
            <ComponentWrapper
              pathSuffix='logout'
              exact
              navbar
              component={Logout}
            />
            <ComponentWrapper
              pathSuffix='login'
              exact
              systemAware
              component={Login}
            />
            <ComponentWrapper
              pathSuffix=''
              systemAware
              exact
              navbar
              component={SystemHomepage}
            />
            <ComponentWrapper
              pathSuffix=''
              eventAware
              exact
              navbar
              component={Homepage}
            />
            <ComponentWrapper
              pathSuffix='fundations'
              eventAware
              exact
              navbar
              component={FundationList}
            />
            <ComponentWrapper
              pathSuffix='users'
              exact
              navbar
              eventAware
              component={UserDashboard}
            />
            <ComponentWrapper
              pathSuffix='transferts'
              eventAware
              exact
              navbar
              component={Transferts}
            />
            <ComponentWrapper
              pathSuffix='fundations/:fundationId(\d+)'
              eventAware
              exact
              navbar
              component={FundationDetails}
            />
            <ComponentWrapper
              pathSuffix='dashboard'
              eventAware
              exact
              navbar
              component={Dashboard}
            />
            <ComponentWrapper
              pathSuffix='support'
              eventAware
              exact
              navbar
              component={Support}
            />
          </Router>
          <ToastContainer />
          <footer className='text-center'>
            <span>
              I used to be a engineer like you, then I took an arrow in the
              knee...
            </span>
            <br />
            <span>&copy;{new Date().getFullYear()} - C.Richard</span>
            <br />
            <span>
              {packagejson.name}@{packagejson.version}
            </span>
          </footer>
        </div>
      </Fragment>
    );
  }
}

export default App;
