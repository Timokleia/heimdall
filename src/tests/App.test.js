import React from "react";
import App from "../App";
import Enzyme, { mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });
describe("Main App", () => {
  it("Render without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
