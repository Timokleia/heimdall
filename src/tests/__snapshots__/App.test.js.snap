// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App Render without crashing 1`] = `
<Fragment
  key="App"
>
  <div
    className="App"
  >
    <BrowserRouter>
      <ComponentWrapper
        component={[Function]}
        exact={true}
        navbar={true}
        pathSuffix="forbiden"
      />
      <ComponentWrapper
        component={[Function]}
        exact={true}
        navbar={true}
        pathSuffix="logout"
      />
      <ComponentWrapper
        component={[Function]}
        exact={true}
        pathSuffix="login"
        systemAware={true}
      />
      <ComponentWrapper
        component={[Function]}
        exact={true}
        navbar={true}
        pathSuffix=""
        systemAware={true}
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix=""
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="fundations"
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="users"
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="transferts"
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="fundations/:fundationId(\\\\d+)"
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="dashboard"
      />
      <ComponentWrapper
        component={[Function]}
        eventAware={true}
        exact={true}
        navbar={true}
        pathSuffix="support"
      />
    </BrowserRouter>
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <footer
      className="text-center"
    >
      <span>
        I used to be a engineer like you, then I took an arrow in the knee...
      </span>
      <br />
      <span>
        Â©
        2021
         - C.Richard
      </span>
      <br />
      <span>
        Heimdall
        @
        3.4.4
      </span>
    </footer>
  </div>
</Fragment>
`;
